x-pack/plugins/log_explorer/common/dataset_selection/single_dataset_selection.ts

```diff
@@ -29,24 +28,19 @@ export class SingleDatasetSelection implements DatasetSelectionStrategy {
   }
 
   toDataviewSpec() {
-    const { name, title } = this.selection.dataset.toDataviewSpec();
-    return {
-      id: this.toURLSelectionId(),
-      name,
-      title,
-    };
+    return this.selection.dataset.toDataviewSpec();
   }
 
-  toURLSelectionId() {
-    return encodeDatasetSelection({
+  toPlainSelection() {
```

Comment:
Good question. I intended to explain this in the description, but didn't get around to it yet :snail: We don't need the data-view-in-id hack anymore, since the `initialState` now gives us more control over the initialization. Additionally, the code located in the log explorer shouldn't have any URL-related knowledge anymore.

The obs_log_explorer app is now responsible of encoding the selection in the URL. For this is doesn't have to put the whole data view in the URL anymore, just the selection. Do you think those savings are insufficient to drop the compression?

Reply:
I see, thanks for clarifying! Regarding the compression, I remember we opted-in for it thinking on the long-term plan of implementing the multi-selection mode in the selector, which we currently don't have.
> For this is doesn't have to put the whole data view

IIRC we were encoding only the dataset selection object and not the whole data view, however the compression benefit is relatively small for single selection mode, feel free to continue dropping it and we can consider to add it back if we notice will be required in future

Reply:
Yes, since the page state in the URL is versioned we are free to make changes to it in the future.